fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -1069,6 +1069,97 @@ def assert_state(self, state: TrainingState) -> None:
             self.training_state == state
         ), f"expected to be in state {state} but current state is {self.training_state}"
 
+    @staticmethod
+    @contextlib.contextmanager
+    def enable_wrap(**kwargs) -> Iterator[None]:
+        """
+        Context manager to wrap modules in FullyShardedDataParallel.
+        Useful for when you'd like to apply the same parameters to all child modules that you wrap.
+
+            Usage::
+
+                with FSDP.enable_wrap(**fully_sharded_params):
+                    # Wraps layer in FSDP if within context
+                    self.l1 = FSDP.wrap(torch.nn.Linear(5, 5))
+                    # Wraps children modules based on bucket_size
+                    self.l2 = FSDP.auto_wrap(TransformerBlock(), bucket_size=1e8)
+
+        """
+        with FullyShardedAutoWrap(**kwargs):
+            yield
+
+    @staticmethod
+    def wrap(module, **kwargs):
+        if FullyShardedAutoWrap.in_autowrap_context:
+            kwargs = {**FullyShardedAutoWrap.kwargs, **kwargs}
+            return FullyShardedDataParallel(module, **kwargs)
+        return module
+
+    @staticmethod
+    def auto_wrap(module, bucket_size=1e8, **kwargs):
+        if FullyShardedAutoWrap.in_autowrap_context:
+            wrapped_module, remainder = FullyShardedAutoWrap.recursive_wrap(module, bucket_size=bucket_size, **kwargs)
+            return wrapped_module
+        return module
+
+
+class FullyShardedAutoWrap:
+    """
+    Helper class to wrap modules based on default config args via a context manager.
+    See ``FullyShardedDataParallel.enable_wrap`` for more information.
+    """
+
+    in_autowrap_context = False
+    kwargs = {}
+
+    def __init__(self, **kwargs):
+        self.kwargs = kwargs
+
+    @staticmethod
+    def enable_autowrap_context(kwargs):
+        FullyShardedAutoWrap.in_autowrap_context = True
+        FullyShardedAutoWrap.kwargs = kwargs
+
+    @staticmethod
+    def disable_autowrap_context():
+        FullyShardedAutoWrap.in_autowrap_context = False
+        FullyShardedAutoWrap.kwargs = {}
+
+    def __enter__(self):
+        self.enable_autowrap_context(self.kwargs)
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.disable_autowrap_context()
+
+    @staticmethod
+    def recursive_wrap(x: nn.Module, bucket_size: Union[int, float], **kwargs) -> Tuple[nn.Module, int]:
```

Comment:
How about renaming to `min_num_params` or something similar, since `bucket` already refers to bucketing of the communication ops and is typically measured in MB.
```suggestion
    def recursive_wrap(x: nn.Module, bucket_size: Union[int, float], **kwargs) -> Tuple[nn.Module, int]:
        """
        Automatically wrap child modules of *x* that meet the given criteria with ``FSDP.wrap``.
        
        Args:
            min_num_params (int): min number of parameters for a child Module to be wrapped
        """
```